sklearn/model_selection/_search.py

```diff
@@ -918,15 +1005,32 @@ class RandomizedSearchCV(BaseSearchCV):
 
     Attributes
     ----------
-    grid_scores_ : list of named tuples
-        Contains scores for all parameter combinations in param_grid.
-        Each entry corresponds to one parameter setting.
-        Each named tuple has the attributes:
-
-            * ``parameters``, a dict of parameter settings
-            * ``mean_validation_score``, the mean score over the
-              cross-validation folds
-            * ``cv_validation_scores``, the list of scores for each fold
+    search_results_ : dict of numpy (masked) ndarrays
```

Comment:
Humm... We plan on adding multiple metric support subsequently. How would `grid_scores_` look for that use case? Do we raise an error when `grid_scores_` is accessed when multiple metrics are required. Or do we return the first metric alone? or all the metrics as a dict of lists (of `_CVScoreTuples`)?

Also the old `grid_search` module is available for all the angry users who can tolerate a `DeprecationWarning`. I have a humble opinion that trying to support old code here will constrain us, not now but in subsequent PRs.